elastic/kibana
# [Controls] [Reporting] Remove controls from reports

Closes https://github.com/elastic/kibana/issues/133547

## Summary

Before this change, controls were **not** included in print optimized reports, but they **were** included in non-print-optimized reports. However, this caused a bug where sometimes the reporting states didn't match - more context can be found [here](https://github.com/elastic/kibana/issues/133547#issuecomment-1151554450). 

To fix this, I removed the `data-shared-items-count` from the `control_group_component`, because it was found that this was causing the ambiguous state for Reporting since it only expects a **single** `data-shared-items-count` for the entire page. This solution removes controls from non-print-optimized reports **entirely**.

Note that, if we wanted to include controls in reports in the future, we would have to pull the `dashboard-shared-items-container` class **outside** to include both the `control_group` and the `DashboardGrid`, with something like:

```typescript
    return (
      <div data-shared-items-container data-title={title} data-description={description}>
        {controlsEnabled && container.getInput().viewMode !== ViewMode.PRINT ? (
          <>
            {isEditMode && panelCount !== 0 && controlGroup?.getPanelCount() === 0 ? (
            ...
        ) : null}
        <div
          data-shared-items-count={panelCount}
          className={useMargins ? 'dshDashboardViewport-withMargins' : 'dshDashboardViewport'}
        >
         ...
        </div>
      </div>
    );
```
However, since we don't have a strong preference for whether or not controls should be visible in non-print-optimized reports anyway, the simplest and most reliable solution is to simply to not include them at all for now.

**How to test:**
1. Create a new dashboard
2. Add at least one control
3. **Don't save the dashboard**
4. Generate a non-print-optimized PDF report
5. Ensure that the report is generated without errors and that controls are not visible in the report

### Checklist
